docs/python_based_backends.md

```diff
@@ -0,0 +1,98 @@
++-------------------------+-------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------+
+| Backend                 | Path                                                        | Config                                                                                                              |
++-------------------------+-------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------+
+| python                  | /opt/tritonserver/backends/python/libtriton_python.so       | {"cmdline":{"auto-complete-config":"true","backend-directory":"/opt/tritonserver/backends","min-compute-capability" |
+|                         |                                                             | :"6.000000","default-max-batch-size":"4"}}                                                                          |
+| my_python_based_backend | /opt/tritonserver/backends/my_python_based_backend/model.py | {"cmdline":{"auto-complete-config":"true","backend-directory":"/opt/tritonserver/backends","min-compute-capability" |
+|                         |                                                             | :"6.000000","default-max-batch-size":"4"}}                                                                          |
++-------------------------+-------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------+
+```
+
+## Background
+
+In some use cases, it is sufficient to implement
+[`TritonPythonModel` interface](https://github.com/triton-inference-server/python_backend#usage)
+only once and re-use it across multiple models. As an example, please refer
+to the [vLLM backend](https://github.com/triton-inference-server/vllm_backend),
+which provides a common python script to serve models supported by vLLM.
+
+Triton Inference Server can handle this special case and treats common
+`model.py` script as a Python-based backend. In the scenario, when model
+relies on a custom Python-based backend, Triton loads `libtriton_python.so`
+first, this ensures that Triton knows how to send requests to the backend
+for execution and the backend knows how to communicate with Triton. Then,
+Triton makes sure to use common `model.py` from the backend's repository,
+and not look for it in the model repository.
+
+While the only required function is `execute`, it is typically helpful
+to enhance your implementation by adding `initialize`, `finalize`,
+and any other helper functions. Users are also encouraged to make use of the
+[`auto_complete_config`](https://github.com/triton-inference-server/python_backend#auto_complete_config)
+function to define standardized input and output properties upfront.
```

Comment:
Would be good to mention which Python interpreter is going to be used and add a link to the Custom Execution environments doc in Python backend so that the user knows how to customize that.

Reply:
Mentioned earlier on L52